ibis/backends/pyspark/__init__.py

```diff
@@ -761,7 +765,6 @@ def _create_cached_table(self, name, expr):
     def _drop_cached_table(self, name):
         self._session.catalog.dropTempView(name)
         t = self._cached_dataframes.pop(name)
-        assert t.is_cached
```

Comment:
I'm not sure why this `assert` fails with spark connect. Identical objects (computed using `id`) have different values of this property across time. That said, I haven't looked into how this property is implemented yet.